x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/search_index/index_view_logic.ts

```diff
@@ -52,36 +69,90 @@ export interface IndicesValues {
 
 export const IndexViewLogic = kea<MakeLogicType<IndicesValues, IndicesActions>>({
   actions: {
+    clearFetchIndexTimeout: true,
+    createNewFetchIndexTimeout: (duration) => ({ duration }),
+    setFetchIndexTimeoutId: (timeoutId) => ({ timeoutId }),
+    startFetchIndexPoll: true,
     startSync: true,
+    stopFetchIndexPoll: true,
   },
   connect: {
-    actions: [StartSyncApiLogic, ['makeRequest', 'apiSuccess', 'apiError']],
+    actions: [
+      StartSyncApiLogic,
+      [
+        'apiError as startSyncApiError',
+        'apiSuccess as startSyncApiSuccess',
+        'makeRequest as makeStartSyncRequest',
+      ],
+      FetchIndexApiLogic,
+      [
+        'apiError as fetchIndexApiError',
+        'apiReset as resetFetchIndexApi',
+        'apiSuccess as fetchIndexApiSuccess',
+        'makeRequest as makeFetchIndexRequest',
+      ],
+    ],
     values: [FetchIndexApiLogic, ['data']],
   },
   listeners: ({ actions, values }) => ({
-    apiError: (e) => flashAPIErrors(e),
-    apiSuccess: () => {
+    createNewFetchIndexTimeout: ({ duration }) => {
+      if (values.fetchIndexTimeoutId) {
+        clearTimeout(values.fetchIndexTimeoutId);
+      }
+      const { indexName } = IndexNameLogic.values;
+      const timeoutId = setTimeout(() => {
+        actions.makeFetchIndexRequest({ indexName });
+      }, duration);
+      actions.setFetchIndexTimeoutId(timeoutId);
+    },
+    fetchIndexApiError: () => {
+      actions.createNewFetchIndexTimeout(FETCH_INDEX_POLLING_DURATION_ON_FAILURE);
+    },
+    fetchIndexApiSuccess: () => {
+      actions.createNewFetchIndexTimeout(FETCH_INDEX_POLLING_DURATION);
+    },
+    makeStartSyncRequest: () => clearFlashMessages(),
+    startFetchIndexPoll: () => {
+      const { indexName } = IndexNameLogic.values;
+      // we rely on listeners for fetchIndexApiError and fetchIndexApiSuccess to handle reccuring polling
+      actions.makeFetchIndexRequest({ indexName });
+    },
+    startSync: () => {
+      if (isConnectorIndex(values.data)) {
+        actions.makeStartSyncRequest({ connectorId: values.data?.connector?.id });
+      }
+    },
+    startSyncApiError: (e) => flashAPIErrors(e),
+    startSyncApiSuccess: () => {
       flashSuccessToast(
         i18n.translate('xpack.enterpriseSearch.content.searchIndex.index.syncSuccess.message', {
           defaultMessage: 'Successfully scheduled a sync, waiting for a connector to pick it up',
         })
       );
     },
-    makeRequest: () => clearFlashMessages(),
-    startSync: () => {
-      if (isConnectorIndex(values.data)) {
-        actions.makeRequest({ connectorId: values.data?.connector?.id });
+    stopFetchIndexPoll: () => {
+      if (values.fetchIndexTimeoutId) {
```

Comment:
this check and clearing the timeout make more sense as a listener on clearFetchIndexTimeOut no?

Reply:
We discussed how I need to do this synchronously rather than relying on a listener